rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -317,22 +332,16 @@ impl<TPayload: Clone + 'static> StreamProcessor<TPayload> {
                 }
 
                 drop(trait_callbacks); // unlock mutex so we can close consumer
-                self.consumer.close();
                 return Err(e);
             }
         }
-        self.shutdown();
         Ok(())
     }
 
     pub fn get_handle(&self) -> ProcessorHandle {
         self.processor_handle.clone()
     }
 
-    pub fn shutdown(&mut self) {
-        self.consumer.close();
-    }
```

Comment:
Maybe having an explicit `fn shutdown(self)` would still be useful at least in tests when you manually call `run_once` in a loop, and you can then `.shutdown()` instead of `drop(…)`

Reply:
Yeah that's fair.